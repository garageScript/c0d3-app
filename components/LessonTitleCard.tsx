import React from 'react'
import NavLink from './NavLink'
import styles from '../scss/lessonTitleCard.module.scss'

export type LessonTitleProps = {
  lessonCoverUrl: string
  lessonUrl: string
  lessonTitle: string
  lessonId: string
  isPassed: boolean
  setShow?: React.Dispatch<React.SetStateAction<boolean>>
  show?: boolean
}

const LessonTitleCard: React.FC<LessonTitleProps> = props => {
  return (
    <div className="card shadow-sm mt-3 col-12 px-0 pt-3 border-0">
      <div className="card-body p-0">
        <div className="d-flex mb-3 px-3">
          <img
            className={`${styles['lessonTitleCard__lesson-cover']} mr-3`}
            src={`/assets/curriculum/${props.lessonCoverUrl}`}
            alt="lesson-cover"
          />
          <div>
            <p className="m-0">
              <NavLink path="/curriculum">Go Back</NavLink>
            </p>
            <h1 className={`${styles['lessonTitleCard__lesson-title']}`}>
              {props.lessonTitle}
            </h1>
          </div>
        </div>
        <div className="card-footer bg-white p-0">
          <NavLink
            path={props.lessonUrl}
            className="btn border-right rounded-0 px-4 py-3"
            external
          >
            LESSON
          </NavLink>
          {/* 768 px is md bootstrap breakpoint */}
          {typeof window !== 'undefined' && window.innerWidth <= 768 ? (
            <div
              onClick={() => props.setShow && props.setShow(!props.show)}
              className="btn border-right rounded-0 px-4 py-3"
            >
              SHOW CHALLENGES
            </div>
          ) : (
            <NavLink
              path={`/curriculum/${props.lessonId}`}
              className="btn border-right rounded-0 px-4 py-3"
            >
              CHALLENGES
            </NavLink>
          )}
          {props.isPassed && (
            <NavLink
              path={`/review/${props.lessonId}`}
              className="btn border-right rounded-0 px-4 py-3"
            >
              REVIEW
            </NavLink>
          )}
        </div>
      </div>
    </div>
  )
}

export default LessonTitleCard
