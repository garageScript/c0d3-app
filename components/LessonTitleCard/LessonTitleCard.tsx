import React from 'react'
import NavLink from '../NavLink'
import styles from './lessonTitleCard.module.scss'
import { useRouter } from 'next/router'
import useBreakpoint from '../../helpers/useBreakpoint'

export type LessonTitleProps = {
  lessonCoverUrl: string
  lessonUrl?: string | null
  lessonSlug: string
  lessonTitle: string
  lessonId: number
  isPassed: boolean
  setShow?: React.Dispatch<React.SetStateAction<boolean>>
  show?: boolean
}

const LessonTitleCard: React.FC<LessonTitleProps> = ({
  lessonCoverUrl,
  lessonUrl,
  lessonSlug,
  lessonTitle,
  isPassed,
  setShow,
  show
}) => {
  const isSmallDownBreakpoint = useBreakpoint('sm', 'down')
  const router = useRouter()

  return (
    <div className="card shadow-sm mt-3 col-12 px-0 pt-3 border-0">
      <div className="card-body p-0">
        <div className="d-flex mb-3 px-3">
          <img
            className={`${styles['lessonTitleCard__lesson-cover']} me-3`}
            src={`/assets/curriculum/${lessonCoverUrl}`}
            alt="lesson-cover"
          />
          <div>
            <p className="m-0">
              <a
                href="#"
                onClick={e => {
                  // Link does not get correct styles without the href
                  // Prevent scroll to top before going back
                  e.preventDefault()
                  router.back()
                }}
                className={`${styles['link']} link-primary`}
              >
                Go Back
              </a>
            </p>
            <h1 className={`${styles['lessonTitleCard__lesson-title']}`}>
              {lessonTitle}
            </h1>
          </div>
        </div>
        <div className="card-footer d-flex bg-white p-0">
          {lessonUrl && (
            <NavLink
              path={lessonUrl}
              className="text-dark d-block border-end rounded-0 px-4 py-3"
            >
              LESSON
            </NavLink>
          )}
          {isSmallDownBreakpoint && setShow ? (
            <div
              onClick={() => setShow!(!show)}
              className="text-dark d-block border-end rounded-0 px-4 py-3"
            >
              SHOW CHALLENGES
            </div>
          ) : (
            <NavLink
              path={`/curriculum/${lessonSlug}`}
              className="text-dark border-end rounded-0 px-4 py-3"
            >
              CHALLENGES
            </NavLink>
          )}
          {isPassed && (
            <NavLink
              path={`/review/${lessonSlug}`}
              className="text-dark d-block border-end rounded-0 px-4 py-3"
            >
              REVIEW
            </NavLink>
          )}
        </div>
      </div>
    </div>
  )
}

export default LessonTitleCard
