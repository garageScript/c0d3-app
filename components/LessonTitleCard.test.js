import React from 'react'
import LessonTitleCard from './LessonTitleCard'
import { render, screen, fireEvent, waitFor } from '@testing-library/react'
describe('LessonTitleCard component', () => {
  const setShow = jest.fn()
  const props = {
    lessonCoverUrl: 'coverUrl',
    lessonUrl: 'lessonUrl',
    lessonTitle: 'Test Lesson',
    lessonId: '0',
    isPassed: true,
    setShow,
    show: false
  }
  test('Should show challenges on click on mobile devices', async () => {
    global.window.innerWidth = 500
    const { container } = render(<LessonTitleCard {...props} />)
    await waitFor(() => fireEvent.click(screen.getByText('SHOW CHALLENGES')))
    expect(setShow).toBeCalledWith(true)
    expect(container).toMatchSnapshot()
  })
  test('Should render default layout for wider screens', async () => {
    global.window.innerWidth = 1080
    const { container } = render(<LessonTitleCard {...props} />)
    expect(container).toMatchSnapshot()
  })
})
