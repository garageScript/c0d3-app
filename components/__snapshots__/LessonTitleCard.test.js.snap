// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LessonTitleCard component Should render default layout for wider screens 1`] = `
<div>
  <div
    class="card shadow-sm mt-3 col-12 px-0 pt-3 border-0"
  >
    <div
      class="card-body p-0"
    >
      <div
        class="d-flex mb-3 px-3"
      >
        <img
          alt="lesson-cover"
          class="lessonTitleCard__lesson-cover mr-3"
          src="/assets/curriculum/coverUrl"
        />
        <div>
          <p
            class="m-0"
          >
            <a
              class=""
              href="/curriculum"
            >
              Go Back
            </a>
          </p>
          <h1
            class="lessonTitleCard__lesson-title"
          >
            Test Lesson
          </h1>
        </div>
      </div>
      <div
        class="card-footer bg-white p-0"
      >
        <a
          class="btn border-right rounded-0 px-4 py-3"
          href="lessonUrl"
          rel="noopener noreferrer"
          target="_blank"
        >
          LESSON
        </a>
        <a
          class="btn border-right rounded-0 px-4 py-3"
          href="/curriculum/0"
        >
          CHALLENGES
        </a>
        <a
          class="btn border-right rounded-0 px-4 py-3"
          href="/review/0"
        >
          REVIEW
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`LessonTitleCard component Should show challenges on click on mobile devices 1`] = `
<div>
  <div
    class="card shadow-sm mt-3 col-12 px-0 pt-3 border-0"
  >
    <div
      class="card-body p-0"
    >
      <div
        class="d-flex mb-3 px-3"
      >
        <img
          alt="lesson-cover"
          class="lessonTitleCard__lesson-cover mr-3"
          src="/assets/curriculum/coverUrl"
        />
        <div>
          <p
            class="m-0"
          >
            <a
              class=""
              href="/curriculum"
            >
              Go Back
            </a>
          </p>
          <h1
            class="lessonTitleCard__lesson-title"
          >
            Test Lesson
          </h1>
        </div>
      </div>
      <div
        class="card-footer bg-white p-0"
      >
        <a
          class="btn border-right rounded-0 px-4 py-3"
          href="lessonUrl"
          rel="noopener noreferrer"
          target="_blank"
        >
          LESSON
        </a>
        <div
          class="btn border-right rounded-0 px-4 py-3"
        >
          SHOW CHALLENGES
        </div>
        <a
          class="btn border-right rounded-0 px-4 py-3"
          href="/review/0"
        >
          REVIEW
        </a>
      </div>
    </div>
  </div>
</div>
`;
